File: 2078. Two Furthest Houses With Different Colors.cpp
class Solution {
public:
    int maxDistance(vector<int>& colors) 
    {
        int i = 0, j = colors.size()-1, ans;
        while(i<=j)
        {
            if(colors[i] != colors[j])
            {
                ans = j-i;
                break;
            }
            j--;
        }
        i = 0;
        while(colors[i] == colors.back()) i++;
        return max(ans, int(colors.size())-i-1);
        return -1;
    }
};

File: 2078. Two Furthest Houses With Different Colors.py
class Solution:
    def maxDistance(self, colors: List[int]) -> int:
        ans = -1
        n = len(colors)
        seen = set()
        for i in range(n):
            if colors[i] in seen:
                continue
            seen.add(colors[i])
            for j in range(n - 1, i, -1):
                if colors[i] != colors[j]:
                    ans = max(ans, abs(i - j))
                    break
        return ans